From 06ba12c4d252c8acc55718fc9d29271ab6bed0c7 Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:58:56 -0400
Subject: [scsi] set correct completion code in scsi_send_eh_cmnd()

Message-id: <1408561176-18133-18-git-send-email-emilne@redhat.com>
Patchwork-id: 88436
O-Subject: [RHEL7.1 PATCH e-stor 17/57] scsi: set correct completion code in scsi_send_eh_cmnd()
Bugzilla: 1132092
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit ac61d19559349e205dad7b5122b281419aa74a82

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit ac61d19559349e205dad7b5122b281419aa74a82
Author: Hannes Reinecke <hare@suse.de>
Date:   Thu May 8 08:09:56 2014 +0200

    scsi: set correct completion code in scsi_send_eh_cmnd()

    ->queuecommand returns '0' for successful command submission,
    so we need to set the correct SCSI midlayer return value
    when calling scsi_log_completion().

    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Reported-by: Robert Elliott <elliott@hp.com>
    Cc: Stephen Cameron <scameron@beardog.cce.hp.com>
    Tested-by: Robert Elliott <elliott@hp.com>
    Signed-off-by: Christoph Hellwig <hch@lst.de>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index 9234327..103d149 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -1029,6 +1029,7 @@ retry:
   rtn = NEEDS_RETRY;
  } else {
   timeleft = wait_for_completion_timeout(&done, timeout);
+  rtn = SUCCESS;
  }
 
  shost->eh_action = NULL;
-- 
1.7.1